virtool/subtractions/data.py

```diff
@@ -83,33 +85,84 @@ async def find(self, find: str, short: bool, ready: bool, query: MultiDictProxy)
                 ).sort("name")
             ]
 
-        data = await paginate(
-            self._mongo.subtraction,
-            db_query,
-            query,
-            base_query={"deleted": False},
-            sort="name",
-            projection=PROJECTION,
-        )
+        data = await self._mongo.subtraction.aggregate(
+            [
+                {
+                    "$facet": {
+                        "total_count": [
+                            {"$count": "total_count"},
+                        ],
+                        "found_count": [
+                            {"$count": "found_count"},
+                        ],
+                        "page_count": [
+                            {"$count": "page_count"},
+                        ],
+                        "per_page": [
+                            {"$count": "per_page"},
+                        ],
+                        "page": [
+                            {"$count": "page"},
+                        ],
+                        "documents": [
+                            {
+                                "$match": query,
+                            },
+                            {"$sort": {"name": 1}},
+                            *lookup_nested_user_by_id(local_field="user.id"),
+                            *lookup_minimal_job_by_id(),
```

Comment:
There are two general important problems here:

1. `$count` merely counts all documents currently in the list. I suggest reading up on what count does and refactoring this code, as most of these are not getting you what your looking for. https://www.mongodb.com/docs/manual/reference/operator/aggregation/count/
2. In the documents facet the it will just return the whole list of subtractions rather than applying any sort of pagination. Looks at the samples find function for an example of how this can be done.